### PR TITLE
Adjusted code to be more generic and improved UI.

### PR DESCRIPTION
**Pull request by jankrzempek**
https://trello.com/c/eSLDiuQk/631-applied-changes-to-parameters

https://trello.com/c/4K1Dt8Cv/635-improve-searching-for-purpleair-and-airbeam-data-to-search-feature

### CODE REVIEW

**Review by SzymonGorowski — APPROVED** [review]
Looks good!

File: AirCasting/SearchAndFollow/MapSessionsDownloader/APIStruct/MapDownloaderMeasurementType.swift [hunk 1]
```diff
@@ -16,6 +16,13 @@ extension MapDownloaderMeasurementType {
         }
     }
     
+    var apiNameSufix: String {
```
Comment (pawelgil) [inline AirCasting/SearchAndFollow/MapSessionsDownloader/APIStruct/MapDownloaderMeasurementType.swift:19]: typo: `Suffix`. It should probably also be renamed to something better... How about `sensorNameSuffix` (see also the prefix name change proposition)

File: AirCasting/SearchAndFollow/MapSessionsDownloader/APIStruct/MapDownloaderSensorType.swift [hunk 2]
```diff
@@ -3,20 +3,18 @@
...
+enum MapDownloaderSensorType: Codable {
     case AB3and2
     case OpenAQ
     case PurpleAir
-    case OzoneSensor
 }
 
 extension MapDownloaderSensorType {
-    var apiName: String {
+    var apiNamePrefix: String {
```
Comment (pawelgil) [inline AirCasting/SearchAndFollow/MapSessionsDownloader/APIStruct/MapDownloaderSensorType.swift:13]: and here maybe `sensorNamePrefix` ?

File: AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift [hunk 3]
```diff
@@ -14,26 +14,69 @@ class SearchViewModel: ObservableObject {
...
+        case Strings.SearchFollowSensorNames.openAQ: return .OpenAQ
+        case Strings.SearchFollowSensorNames.purpleAir: return .PurpleAir
+        case Strings.SearchFollowSensorNames.openAQOzone: return .OpenAQ
+        default: return nil
+        }
     }
     
-    func onOzoneSensorTap(with sensor: OzoneSensorType) {
-        self.sensor = MapDownloaderSensorType.allCases.first(where: { $0.capitalizedName == sensor.capitalizedName }) ?? .OzoneSensor 
+    func onParameterTap(with param: String) {
```
Comment (pawelgil) [inline AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift:60]: a bit messy structure in this file. Please move all `public` methods above `private` ones to improve readability (and do the same with properties)

File: AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift [hunk 4]
```diff
@@ -14,26 +14,69 @@ class SearchViewModel: ObservableObject {
     @Published var addressName = ""
     @Published var addresslocation = CLLocationCoordinate2D(latitude: 20.0, longitude: 20.0)
     @Published var alert: AlertInfo?
-    @Published private var parameter: MapDownloaderMeasurementType = .ozone
-    @Published private var sensor: MapDownloaderSensorType = .OzoneSensor
+    @Published var measurementTypes = [SearchParameter]()
+    @Published var sensorTypes = [SearchParameter]()
+    
+    init() {
```
Comment (pawelgil) [inline AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift:20]: move init between properties and functions (this is the most followed convention)

**Review by pawelgil — CHANGES_REQUESTED** [review]
small styling stuff needed

**Review by pawelgil — APPROVED** [review]

**Pull request merged**